File: bindings/c/src/wave_transmission.f90
module wave_transmission_c
implicit none
private

contains

!!!###################################################################################

subroutine evaluate_wave_transmission_c(n_time,a0,no_freq,a,b,n_bcparams,bc_params) bind(C, name="evaluate_wave_transmission_c")

use wave_transmission, only: evaluate_wave_transmission
use arrays,only:dp
implicit none
integer, intent(in) :: n_time
integer,intent(in) :: no_freq
real(dp),intent(in) :: a0
real(dp),intent(in) :: a(no_freq),b(no_freq)
integer, intent(in) :: n_bcparams
real(dp), intent(in) :: bc_params(n_bcparams)

#if defined _WIN32 && defined __INTEL_COMPILER
call so_evaluate_wave_transmission(n_time,a0,no_freq,a,b,n_bcparams,bc_params)
#else
call evaluate_wave_transmission(n_time,a0,no_freq,a,b,n_bcparams,bc_params)
#endif

end subroutine evaluate_wave_transmission_c

!###################################################################################
end module wave_transmission_c

File: bindings/c/src/wave_transmission.c
#include "wave_transmission.h"

void evaluate_wave_transmission_c(int *n_time, double *heartrate, double *a0, int *no_freq, double *a, double *b, int *n_bcparams, double *bc_params);

void evaluate_wave_transmission(int n_time, double heartrate, double a0, int no_freq, double *a, double *b, int n_bcparams, double *bc_params)
{
    evaluate_wave_transmission_c(&n_time,&heartrate, &a0, &no_freq, a, b, &n_bcparams,bc_params);
}
